consensus/safety-rules/src/remote_service.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    network::{NetworkClient, NetworkServer},
+    persistent_storage::PersistentStorage,
+    serializer::{SafetyRulesInput, SerializerClient, SerializerService, TSerializerClient},
+    Error, SafetyRules,
+};
+use consensus_types::common::Payload;
+use libra_types::crypto_proxies::ValidatorSigner;
+use std::{marker::PhantomData, net::SocketAddr, sync::Arc};
+
+pub trait RemoteService<T: Payload> {
+    fn client(&self) -> SerializerClient<T> {
+        let network_client = NetworkClient::connect(self.server_address()).unwrap();
+        let service = Box::new(RemoteClient::new(network_client));
+        SerializerClient::new_client(service)
+    }
+
+    fn server_address(&self) -> SocketAddr;
+}
+
+pub fn execute<T: Payload>(
+    storage: Box<dyn PersistentStorage>,
```

Comment:
we probably need different name for safety rule persistent storage and consensus one, PersistentSafetyStorage vs PersistentLivenessStorage?